### ISSUE TITLE
[android] BUILD FAILED

### ISSUE THREAD

**Issue by studigen** [expo/expo#22558]
### Summary

when run script with `expo run:android` or `eas build --platform android --local`, occur build fail.

```
✔ Created native project | gitignore skipped
✔ Added Metro config
✔ Updated package.json and added index.js entry point for iOS and Android
› Removed "main": "node_modules/expo/AppEntry.js" from package.json because we recommend using index.js as main instead

› Installing using yarn
> yarn install
Using node to generate images. This is much slower than using native packages.
› Optionally you can stop the process and try again after successfully running `npm install -g sharp-cli`.

» android: userInterfaceStyle: Install expo-system-ui in your project to enable this feature.
✔ Config synced
› Building app...
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'my-app'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':classpath'.
      > Could not resolve com.android.tools.build:gradle:7.4.1.
        Required by:
            project :
         > No matching variant of com.android.tools.build:gradle:7.4.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.5.1' but:
             - Variant 'apiElements' capability com.android.tools.build:gradle:7.4.1 declares a library, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'javadocElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a component, and its dependencies declared externally:
                 - Incompatible because this component declares documentation and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java version (required compatibility with Java 8)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'runtimeElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'sourcesElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a component, and its dependencies declared externally:
                 - Incompatible because this component declares documentation and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java version (required compatibility with Java 8)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
      > Could not resolve com.android.tools.build:gradle:7.3.1.
        Required by:
            project : > project :react-native-gradle-plugin
         > No matching variant of com.android.tools.build:gradle:7.4.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.5.1' but:
             - Variant 'apiElements' capability com.android.tools.build:gradle:7.4.1 declares a library, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'javadocElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a component, and its dependencies declared externally:
                 - Incompatible because this component declares documentation and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java version (required compatibility with Java 8)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'runtimeElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')
             - Variant 'sourcesElements' capability com.android.tools.build:gradle:7.4.1 declares a runtime of a component, and its dependencies declared externally:
                 - Incompatible because this component declares documentation and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java version (required compatibility with Java 8)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '7.5.1')

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
```

### What platform(s) does this occur on?

Android

### SDK Version

48.0.15

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v18.14.0/bin/yarn
      npm: 9.3.1 - ~/.nvm/versions/node/v18.14.0/bin/npm
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
      Android SDK:
        API Levels: 23, 26, 28, 29, 30
        Build Tools: 28.0.3, 34.0.0
        System Images: android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.9971841
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 3.12.1
    Expo Workflow: managed

### Minimal reproducible example

it can reproduce on new project created with `npx create-expo-app my-app`

**Comment by Kim-Morvan** [comment]
I have exactly the same problem when running npx expo run:android.
Works perfectly with  npx expo run:ios

**Comment by brentvatne** [comment]
https://stackoverflow.com/questions/75114728/no-matching-variant-of-com-android-tools-buildgradle7-4-0-was-found

**Comment by lucacri** [comment]
I'm still facing this issue, even when I simply send it to expo's servers to build it. I tried to understand from the link you sent, but it doesn't seem to work for me. Any idea/suggestion?

**Comment by nataliepan** [comment]
same also getting this issue

**Comment by Jose-cd** [comment]
facing the same issue

**Comment by Jose-cd** [comment]
@nataliepan @lucacri  @Kim-Morvan @studigen I fixed it by installing Java JDK version 11 on my mac.
<img width="813" alt="Screenshot 2023-06-30 at 11 35 39 PM" src="https://github.com/expo/expo/assets/71904320/fbe1ee5c-7ba9-41f9-ac08-5697d5a7a2ae">

**Comment by markanthonyuy** [comment]
Installing Java JDK v17 works on my mac

<img width="1291" alt="Screenshot 2023-07-01 at 9 38 04 PM" src="https://github.com/expo/expo/assets/4565996/ff8f8be5-49df-4197-a45d-ac508f21ff24">

**Comment by tholmgren** [comment]
Check to make sure your namespace and packages names are correct.
Here's a great video, if that's your issue.
https://www.youtube.com/watch?v=TSfE-4CVgDU

**Comment by aremu-smog** [comment]
> @nataliepan @lucacri @Kim-Morvan @studigen I fixed it by installing Java JDK version 11 on my mac. <img alt="Screenshot 2023-06-30 at 11 35 39 PM" width="813" src="https://user-images.githubusercontent.com/71904320/250233068-fbe1ee5c-7ba9-41f9-ac08-5697d5a7a2ae.png">

This worked for me.

**Comment by KabirSinghShekhawat** [comment]
> > @nataliepan @lucacri @Kim-Morvan @studigen I fixed it by installing Java JDK version 11 on my mac. <img alt="Screenshot 2023-06-30 at 11 35 39 PM" width="813" src="https://user-images.githubusercontent.com/71904320/250233068-fbe1ee5c-7ba9-41f9-ac08-5697d5a7a2ae.png">
> 
> This worked for me.

Damn, after spending all this time on this issue, all I had to do was install a JDK lol. I had a hunch that the issue is JDK related but expected android studio to detect if Java is not installed.

**Comment by BrodaNoel** [comment]
Worked after downloading Java JDK 11

**Comment by AlbertNewtonKEMOGNE** [comment]
I fixed it installing jdk  11.0.2 on my Windows 10. Link : https://www.codejava.net/java-se/download-and-install-java-11-openjdk-and-oracle-jdk
![expoJDK11](https://github.com/expo/expo/assets/143473674/33528deb-c062-4b7e-aafc-8069f81cef99)
